Resync screen-orientation tests from upstream WPT

cdumez opened the pull request:
#### 32793ec47fe34a49cebf9b701cf1356e5c857ad4
<pre>
Resync screen-orientation tests from upstream WPT
<a href="https://bugs.webkit.org/show_bug.cgi?id=248624">https://bugs.webkit.org/show_bug.cgi?id=248624</a>

Reviewed by Tim Nguyen.

Resync screen-orientation tests from upstream WPT b763d6d0c6e3ede36.

* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/META.yml:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/hidden_document.html:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/orientation-reading.html:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/resources/orientation-utils.js:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/resources/sandboxed-iframe-locking.html:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/w3c-import.log:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/screen-orientation/lock-sandboxed-iframe-expected.txt:
* LayoutTests/tests-options.json:

Canonical link: <a href="https://commits.webkit.org/257476@main">https://commits.webkit.org/257476@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/66e97f091668eee9408fc788abef291848beda24

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/99048 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/8251 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/32193 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/108445 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/168694 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/8798 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/85614 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/91562 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/106399 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/104772 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/6661 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/90240 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/33679 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/88483 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/21585 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/76536 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/2149 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/23101 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/2047 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/45506 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/7012 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/42576 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2608 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/3464 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

marcoscaceres commented on LayoutTests/imported/w3c/web-platform-tests/screen-orientation/resources/sandboxed-iframe-locking.html:
```diff
@@ -20,12 +20,7 @@
-    data.result = "errored";
-    data.name = "fullscreen exit failed";
-  }
+  await document.exitFullscreen();
```
Oh, whoops! This is a regression (probably my fault 😰) . The try/catch is needed here, otherwise it TIMEOUTs below.

nt1m replied:
This needs to be fixed upstream. Then reimported in this patch

cdumez replied:
Fixing via https://github.com/web-platform-tests/wpt/pull/37301.

marcoscaceres reviewed (changes requested):
Possible regression causing a TIMEOUT.

cdumez committed:
Resync screen-orientation tests from upstream WPT
https://bugs.webkit.org/show_bug.cgi?id=248624

Reviewed by Tim Nguyen.

Resync screen-orientation tests from upstream WPT b763d6d0c6e3ede36.

* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/META.yml:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/hidden_document.html:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/orientation-reading.html:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/resources/orientation-utils.js:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/resources/sandboxed-iframe-locking.html:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/w3c-import.log:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/screen-orientation/lock-sandboxed-iframe-expected.txt:
* LayoutTests/tests-options.json:

Canonical link: https://commits.webkit.org/257476@main
